tests/integration/targets/dcnm_service_node/tests/dcnm/deleted.yaml

```diff
@@ -0,0 +1,315 @@
+##############################################
+##                 SETUP                    ##
+##############################################
+
+- name: DELETED - Verify if fabric is deployed.
+  cisco.dcnm.dcnm_rest:
+    method: GET
+    path: /rest/control/fabrics/{{ ansible_it_fabric }}
+  register: result
+
+- assert:
+    that:
+    - 'result.response.DATA != None'
+
+- name: DELETED - Verify if service fabric is deployed.
+  cisco.dcnm.dcnm_rest:
+    method: GET
+    path: /rest/control/fabrics/{{ ansible_ext_fabric }}
+  register: result
+
+- assert:
+    that:
+    - 'result.response.DATA != None'
+
+- name: DELETED - Clean up any existing service node
+  cisco.dcnm.dcnm_service_node:
+    fabric: "{{ ansible_it_fabric }}"
+    service_fabric: "{{ ansible_ext_fabric }}"
+    state: deleted
+
+- name: DELETED - sleep for 40 seconds for DCNM to completely update the state
```

Comment:
Instead of just sleeping for `40` seconds I would prefer that we use the Ansible until/retries/loop construct.  Please also examine other sleep statements to see if we can do something like this instead so we don't sleep longer then needed

In this particular case you could possibly loop until `'result.response.DATA == None'`

Example:

```yaml
  - name: Query fabric state until vrfStatus transitions to DEPLOYED state
    cisco.dcnm.dcnm_vrf:
      fabric: "{{ fabric_name }}"
      state: query
    register: result
    until:
      - "result.response[0].parent.vrfStatus is search('DEPLOYED')"
      - "result.response[1].parent.vrfStatus is search('DEPLOYED')"
      - "result.response[2].parent.vrfStatus is search('DEPLOYED')"
    retries: 20
    delay: 5
```

Reply:
Addressed the same in whatever place this can be applied, was able to reduce the run time considerably.